src/status_im/data_store/realm/core.cljs

```diff
@@ -84,11 +84,26 @@
 (defn write [realm f]
   (.write realm f))
 
+(defn- prepare-for-transit 
+  "Following Transit documentation, escape leading ~'s
+  in strings by prepending a ~. This prepares for subsequent 
+  fetching from Realm where Transit is used for JSON parsing." 
+  [message]
+  (letfn [(walk-fn [e]
+            (if (and (string? e)
+                     (first e)
+                     (contains? #{"~" "^" "`"} (first e))
+                     ) 
+              (str "~" e) 
+              (walk/walk walk-fn identity e)))]
+    (walk/walk walk-fn identity message)))
```

Comment:
We could use `postwalk`, then no need to use recursion inside `walk-fn`. 
Something like
```clojure
(letfn [(walk-fn [e]
            (cond->> e
                     (to-be-escaped? e)
                     (str escape-char)))]
    (walk/postwalk walk-fn message))
```
But this is just a suggestion